sdk/src/Core/AWSXRayRecorder.netcore.cs

```diff
@@ -151,9 +157,12 @@ public static AWSXRayRecorder Instance
         {
             get
             {
-                if (_instance == null)
+                lock (_instance_lock)
```

Comment:
I would prefer avoiding lock here since this is in hot code path. Almost all X-Ray related operations call `Instance`. Here instead of lock, we can do an optimization. 
`_instance` is static instance. So we can initialize `_instance = new AWSXRayRecorderBuilder().Build();` [here](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/src/Core/AWSXRayRecorder.netcore.cs#L35) and eliminate ` if (_instance == null)` on the `get` of `Instance`.

Something similar to currently we do in `net45` platform. [Reference1](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/src/Core/AWSXRayRecorder.cs#L69), [Reference2](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/src/Core/AWSXRayRecorderImpl.cs#L40)

If you can modify this PR accordingly and do a testing on your environment, this will be helpful.
Avoiding locks will help in maintaining the current performance.

Reply:
Perfect will post an update today.

Reply:
Updated the pull request, this seems to solve it as well.

Reply:
Looks like unit tests fail, taking a look today.

Reply:
That's resolved now.